worker/uploader.go

```diff
@@ -158,23 +169,28 @@ func handleSectorUpload(uploadErr error, uploadDuration, totalDuration time.Dura
 		return true, false, float64(ms), float64(rhpv2.SectorSize / ms)
 	}
 
+	// upload failed because we weren't able to create a payment, in this case
+	// we want to punish the host but only to ensure we stop using it, meaning
+	// we don't increment consecutive failures
+	if utils.IsErr(uploadErr, errFailedToCreatePayment) {
+		return false, false, float64(time.Hour.Milliseconds()), 0
+	}
+
 	// upload failed because the sector was already uploaded by another host, in
 	// this case we want to punish the host for being too slow but only when we
 	// weren't overdriving or when it took too long to dial
-	if errors.Is(uploadErr, errSectorUploadFinished) {
-		slowDial := errors.Is(uploadErr, errDialTransport) && totalDuration > time.Second
-		if !overdrive || slowDial {
+	if utils.IsErr(uploadErr, errSectorUploadFinished) {
+		slowDial := utils.IsErr(uploadErr, errDialTransport) && totalDuration > time.Second
+		slowLock := utils.IsErr(uploadErr, errAcquireContractFailed) && totalDuration > time.Second
+		slowFetchRev := utils.IsErr(uploadErr, errFetchRevisionFailed) && totalDuration > time.Second
+		if !overdrive || slowDial || slowLock || slowFetchRev {
 			failure = overdrive
 			uploadEstimateMS = float64(totalDuration.Milliseconds() * 10)
-			logger.Debugw("sector upload failure was penalised", "uploadError", uploadErr, "uploadDuration", uploadDuration, "totalDuration", totalDuration, "overdrive", overdrive, "penalty", totalDuration.Milliseconds()*10)
-		} else {
-			logger.Debugw("sector upload failure was ignored", "uploadError", uploadErr, "uploadDuration", uploadDuration, "totalDuration", totalDuration, "overdrive", overdrive)
 		}
 		return false, failure, uploadEstimateMS, 0
 	}
 
 	// in all other cases we want to punish the host for failing the upload
-	logger.Debugw("sector upload failure was penalised", "uploadError", uploadErr, "uploadDuration", uploadDuration, "totalDuration", totalDuration, "overdrive", overdrive, "penalty", time.Hour)
```

Comment:
Why get rid of all the debug logging?

Reply:
I pulled it out of this method to avoid having to pass in whatever we want to add to this debug log. I ran into it trying to add the host key and had to pass it in only to log it... This also avoids passing in the logger and I felt we have everything we need on the caller side.